### PR TITLE
#2366 remove java 9 specific documentation

### PR DESCRIPTION
**Pull request by JudeNiroshan**
### Fixes: #2366

Removed Java 9 specific documentation as we are already in Java 15 and Java 11 LTS.

### CODE REVIEW

File: documentation/src/main/asciidoc/chapter-2-set-up.asciidoc [hunk 1]
```diff
@@ -248,13 +248,11 @@ If a policy is given for a specific mapper via `@Mapper#unmappedTargetPolicy()`,
 |`WARN`
 |===
 
-=== Using MapStruct on Java 9
+=== Using MapStruct on Java 9 and higher versions
```
Comment (gunnarmorling) [inline documentation/src/main/asciidoc/chapter-2-set-up.asciidoc:251]: I'd suggest "Using MapStruct with the Java Module System"

File: documentation/src/main/asciidoc/chapter-2-set-up.asciidoc [hunk 2]
```diff
@@ -248,13 +248,11 @@ If a policy is given for a specific mapper via `@Mapper#unmappedTargetPolicy()`,
...
 
-MapStruct can be used with Java 9 (JPMS), support for it is experimental.
+MapStruct can be used with Java 9 and higher versions.
 
-A core theme of Java 9 is the modularization of the JDK. One effect of this is that a specific module needs	to be enabled for a project in order to use the `javax.annotation.Generated` annotation. `@Generated` is added by MapStruct to generated mapper classes to tag them as generated code, stating the date of generation, the generator version etc.
-
-To allow usage of the `@Generated` annotation the module _java.xml.ws.annotation_ must be enabled. When using Maven, this can be done like this:
+To allow usage of the `@Generated` annotation the module _java.xml.ws.annotation_ can be enabled. When using Maven, this can be done like this:
```
Comment (gunnarmorling) [inline documentation/src/main/asciidoc/chapter-2-set-up.asciidoc:255]: The `javax.annotation.processing.Generated` annotation should be mentioned instead (from the jdk.compiler) module.

**Review by gunnarmorling — COMMENTED** [review]
Thx! Some comments inline.

File: documentation/src/main/asciidoc/chapter-2-set-up.asciidoc (hunk 1 above)
Comment (JudeNiroshan) [reply documentation/src/main/asciidoc/chapter-2-set-up.asciidoc]: 6510ece

File: documentation/src/main/asciidoc/chapter-2-set-up.asciidoc (hunk 2 above)
Comment (JudeNiroshan) [reply documentation/src/main/asciidoc/chapter-2-set-up.asciidoc]: 5d09733

**Comment by filiphr** [comment]
LGTM. Merging it now. Thanks @JudeNiroshan.

@gunnarmorling let me know if you have more remarks and we can update it

**Pull request merged**